keps/sig-cli/kustomize-secret-generator-plugins.md

```diff
@@ -28,33 +28,59 @@ status: implementable
     * [Goals](#goals)
     * [Non-Goals](#non-goals)
 * [Proposal](#proposal)
-    * [Risks and Mitigations](#risks-and-mitigations)
+* [Risks and Mitigations](#risks-and-mitigations)
 * [Graduation Criteria](#graduation-criteria)
 * [Implementation History](#implementation-history)
 
 ## Summary
 
-Kustomize wants to generate Kubernetes Secret objects from general key-value pairs where the value is supposed to be a secret. Currently Kustomize only supports reading secret values from local files which raises security concerns about file lifetime and access. Reading secret values from the execution of arbitrary "commands" in a kustomization.yaml file introduces concerns in a world where kustomization files can be used directly from the internet when a user runs `kustomize build`.
-
-This proposal describes obtaining secret values safely using golang [plugins](https://golang.org/pkg/plugin/).
+Kustomize wants to generate Kubernetes Secret objects from
+general key:value pairs where the value is supposed to be a
+secret. Currently Kustomize only supports reading secret
+values from local files which raises security concerns about
+file lifetime and access. Reading secret values from the
+execution of arbitrary "commands" in a kustomization.yaml
+file introduces concerns in a world where kustomization
+files can be used directly from the internet when a user
+runs `kustomize build`.  This proposal describes the syntax
+for a new key:value generator plugin framework supporting an
+arbitrary number of plugin types to generate key:value
+pairs.
 
 ## Motivation
 
-Not having a way to integrate `SecretGenerator` with secret management tools requires hacky workarounds.
+Not having a way to integrate `SecretGenerator` with
+secret management tools requires hacky, insecure workarounds.
 
 ### Goals
 
-- Give users a way to intergrate `SecretGenerator` with secret management tools in a safe way.
+- In the `GeneratorArgs` section of a kustomization
+  file, a user may specify a plugin type, and a
+  specific instance of that type, for generating
+  key:value pairs.
+
+- The specification will allow for any number of plugin
+  types, and any number of instances of those types.
+  
+- The first type supported will be
+  [goplugins](https://golang.org/pkg/plugin), to
+  support a user comfortable with their limitations
+  (see below) and to initially exercise the framework.
+
+- Other kinds of plugins, e.g. an _execute this binary_
+  plugin, should be subsequently easy to add.
+
 
 ### Non-Goals
 
-- Kustomize will not handle plugin installation/management.
+- Kustomize will not handle key:value generation plugin
```

Comment:
Support for plugins that are not included as part of a distribution and built on the same host / go compiler / transitive go libraries / etc as the host program.

Reply:
added comment